### DISCUSSION TITLE
MPD Volume control: Mixer Device <> MPD Software mixed up

### DISCUSSION THREAD

**Discussion by holzboa** [rern/rAudio#193] [Fixed Issues]
I use Raspberry Pi 2 Model B & HifiBerry Amp2 so "MPD Software" was the way to control volume in previous versions.
I just installed the new version and recognized that using the "MPD Software Volume Control" doesn't change the volume. 
Instead i had to switch to "Mixer Device" to achieve the same functionality so maybe preconfigured settings are mixed up.

Otherwise the new version looks and works great - thanks a lot!

**Comment by rern** [comment]
Try update: Settings > Addons > rAudio - Update

**Reply by holzboa** [reply]
Updating was the first step after installing.

**Reply by rern** [reply]
Then:
```sh
aplay -l
amixer -c 0 scontrols
amixer -c 1 scontrols
mpd-data.sh
```

**Comment by holzboa** [comment]
# aplay -l

**** List of PLAYBACK Hardware Devices ****
card 0: sndrpihifiberry [snd_rpi_hifiberry_dacplus], device 0: HiFiBerry DAC+ HiFi pcm512x-hifi-0 [HiFiBerry DAC+ HiFi pcm512x-hifi-0]
  Subdevices: 1/1
  Subdevice #0: subdevice #0


# amixer -c 0 scontrols

Simple mixer control 'DSP Program',0
Simple mixer control 'Analogue',0
Simple mixer control 'Analogue Playback Boost',0
Simple mixer control 'Auto Mute',0
Simple mixer control 'Auto Mute Mono',0
Simple mixer control 'Auto Mute Time Left',0
Simple mixer control 'Auto Mute Time Right',0
Simple mixer control 'Clock Missing Period',0
Simple mixer control 'Deemphasis',0
Simple mixer control 'Digital',0
Simple mixer control 'Max Overclock DAC',0
Simple mixer control 'Max Overclock DSP',0
Simple mixer control 'Max Overclock PLL',0
Simple mixer control 'Volume Ramp Down Emergency Rate',0
Simple mixer control 'Volume Ramp Down Emergency Step',0
Simple mixer control 'Volume Ramp Down Rate',0
Simple mixer control 'Volume Ramp Down Step',0
Simple mixer control 'Volume Ramp Up Rate',0
Simple mixer control 'Volume Ramp Up Step',0

# amixer -c 1 scontrols

Invalid card number.


# mpd-data.sh

{ "devices" : [{ "aplayname" : "hifiberry-dacplus" , "card" : 0 , "device" : 0 , "dop" : 0 , "hw" : "hw:0,0" , "hwmixer" : "Digital" , "mixers" : 1 , "mixerdevices" : ["Digital"] , "mixermanual" : false , "mixertype" : "hardware" , "name" : "HiFiBerry Amp2" }] , "active" : true , "asoundcard" : 0 , "audioaplayname" : "hifiberry-dacplus" , "audiooutput" : "HiFiBerry Amp2" , "autoupdate" : false , "buffer" : false , "bufferval" : false , "bufferoutput" : true , "bufferoutputval" : 16384 , "counts" : { "album": 0, "albumartist": 5543, "artist": 21946, "composer": 614, "date": 83, "genre": 214, "nas": 3, "sd": 0, "usb": 0, "song": 147425, "webradio": 0 } , "crossfade" : false , "crossfadeval" : false , "custom" : false , "ffmpeg" : true , "normalization" : false , "reboot" : "" , "replaygain" : false , "replaygainval" : "" , "soxr" : false , "soxrval" : "" , "version" : "0.22.4-2" }

**Reply by rern** [reply]
- Setup Volume Control to MPD Software
- Commands:
  - `mpc volume`
  - `mpc volume 50`
- Playback page
  - Volume knob should be at 50
  - Set Volume knob to 30
- Command:
  - `mpc volume`
  - A result should show `Volume: 30%`

**Reply by holzboa** [reply]
Thanks, that works. 
Now I can use 2 settings to control volume but isn't  "Mixer Device" made for hardware volume control?

**Reply by rern** [reply]
**Output**
- **Device** - Audio hardware devices, on-board, I2S or USB
- **Mixer Device** - Available mixers(on-board volume controls) of the current **Device**
- **Volume Control** - Controls to use for volume level of the current **Device**

Your DAC seems to have a mixer device listed. It's better than **MPD Software**.

**Reply by holzboa** [reply]
Thanks for pointing that out - strange that "Mixer Device" didn't work in r5- or r6-edition with the HifiBerry Amp2.
